src/main/java/me/ryanhamshire/GriefPrevention/PlayerEventHandler.java

```diff
@@ -1135,10 +1135,15 @@ public void onPlayerInteractEntity(PlayerInteractEntityEvent event)
         //always allow interactions when player is in ignore claims mode
         if (playerData.ignoreClaims) return;
 
-        //don't allow container access during pvp combat
+        //check allow container access during pvp combat
         if ((entity instanceof StorageMinecart || entity instanceof PoweredMinecart))
         {
-            if (playerData.inPvpCombat())
+            //allow the access if the entity is in a claim AND the user has access to the claim AND container access is allowed during PvP.
+            Claim claim = this.dataStore.getClaimAt(entity.getLocation(), false, null);
+            Supplier<String> noContainersReason = claim.checkPermission(player, ClaimPermission.Inventory, event);
+            boolean allowcontainerAccess = claim != null && noContainersReason == null && instance.config_pvp_allowTrustedContainerAccess; 
+            
+            if (!allowcontainerAccess && playerData.inPvpCombat())
```

Comment:
`claim` might be null (in wilderness), so we need to check for `claim != null` before we check permissions to avoid NPE, something like:

```suggestion
            Claim claim = this.dataStore.getClaimAt(entity.getLocation(), false, null);
            boolean allowContainerAccess = false;
            if (claim != null) {
              Supplier<String> noContainersReason = claim.checkPermission(player, ClaimPermission.Inventory, event);
              allowContainerAccess = noContainersReason == null && instance.config_pvp_allowTrustedContainerAccess;
            }

            if (!allowContainerAccess && playerData.inPvpCombat())
```

(also `allowcontainerAccess` → `allowContainerAccess`)